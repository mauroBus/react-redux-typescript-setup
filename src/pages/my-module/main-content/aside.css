@import "../../../styles/variables";

.aside {
  border: solid var(--blue-grey);
  float: right;
  height: 100%;
  width: 12rem;
  padding: 1rem;
  position: absolute;
  right: 0;
  top: 0;
}

.list {
  list-style-type: none;
  padding: 0;
}

.link {
  text-decoration: none;
}
