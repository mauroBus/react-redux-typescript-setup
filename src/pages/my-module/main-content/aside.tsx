import React from 'react'
import t from 'format-message'
import styles from './aside.css'

export const Aside = () => (
  <aside className={styles.aside}>
    <p>{t('Aside')}</p>
    <nav data-qa="aside-nav">
      <ul className={styles.list}>
        {[...new Array(4)].map((_, index) => (
          <li key={index} data-qa="aside-nav-link">
            <a className={styles.link} href="/">
              {t('Link {nbr, number}', { nbr: index + 1 })}
            </a>
          </li>
        ))}
      </ul>
    </nav>
  </aside>
)
