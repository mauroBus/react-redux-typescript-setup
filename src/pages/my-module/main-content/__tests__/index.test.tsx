import { shallow } from 'enzyme'
import React from 'react'
import { ITEMS } from '../../list.example'
import { getUniqueID } from '../../utils'
import { Aside } from '../aside'
import { MainContent } from '../'

const mockedUUID = 'mocked-unique-ID'
jest.mock('../../utils', () => ({
  getUniqueID: jest.fn(() => mockedUUID),
}))

describe('MyModule::MainContent', () => {
  const defaults = {
    items: ITEMS,
    methodA() {},
    methodB() {},
  }
  const addBtn = '[data-qa="main-content-add-item-btn"]'

  beforeEach(() => {
    ;(getUniqueID as jest.Mock).mockClear()
  })

  it('renders as a "main" element', () => {
    const tree = shallow(<MainContent {...defaults} />)
    expect(tree.type()).toBe('main')
  })

  it('renders a couple of articles', () => {
    const tree = shallow(<MainContent {...defaults} />)
    const articles = tree.find('article')
    expect(articles).toHaveLength(2)
  })

  it('renders the first article', () => {
    const tree = shallow(<MainContent {...defaults} />)

    const article = tree.find('article').at(0)
    const firstChild = article.childAt(0)
    const secondChild = article.childAt(1)

    expect(firstChild.text()).toBe('Article')
    expect(firstChild.type()).toBe('h1')
    expect(secondChild.text()).toBe(
      'I am text in article, also there can be an image:'
    )
  })

  it('renders the second article', () => {
    const tree = shallow(<MainContent {...defaults} />)
    const article = tree.find('article').at(1)
    expect(article).toMatchSnapshot()
  })

  it('calls "methodA" on button click', () => {
    const methodA = jest.fn()
    const tree = shallow(<MainContent {...defaults} methodA={methodA} />)
    tree.find(addBtn).simulate('click')
    expect(methodA).toHaveBeenCalledWith(`item-${mockedUUID}`)
  })

  it('instantiates the Aside component', () => {
    const tree = shallow(<MainContent {...defaults} />)
    expect(tree.find(Aside).exists()).toBe(true)
  })
})
