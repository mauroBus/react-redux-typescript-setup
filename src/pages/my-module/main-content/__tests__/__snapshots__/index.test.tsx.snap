// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyModule::MainContent renders the second article 1`] = `
<article
  className="article"
>
  <h2>
    Lorem ipsum
  </h2>
  <section>
    <ul>
      <Item
        id="1"
        key="1"
        text="Item 1"
      />
      <Item
        id="2"
        key="2"
        text="Item 2"
      />
      <Item
        id="3"
        key="3"
        text="Item 3"
      />
      <Item
        id="4"
        key="4"
        text="Item 4"
      />
    </ul>
  </section>
  <section>
    <Button
      data-qa="main-content-add-item-btn"
      onClick={[Function]}
    >
      Add an item
    </Button>
  </section>
</article>
`;
